wordpress-mobile/WordPress-iOS
# Activity Log: mark discarded events client side and show Jetpack icon for Jetpack actions

**Another step for** #7832 

In this PR we are marking the discarded events client side. Discarded events are non-rewindable and having a way to identify them and displaying differently will help in our next steps.
For now we are processing the entire stream as we receive it, since we do no paging whatsoever. In the future the attribute `isDiscarded` should come server side and this code would only be used to display how a rewind would change the events status.

Documentation on the subject can be found [here](https://github.com/Automattic/wp-calypso/tree/master/client/state/activity-log/log/is-discarded) 

We've also added the Jetpack icon as a gravatar when the action was performed by Jetpack.

**To test:**

Make sure https://github.com/Automattic/wp-calypso/pull/20070 is merged.

On a site with ActivityLog enabled and some restores done (for example https://betterjetpackxp.mystagingwebsite.com) check that the events that are black and white on Calypso are grayed out in the app.

 Discarded events|Non discarded events
-|-
![simulator screen shot - iphone 8 - 2017-11-22 at 16 14 52](https://user-images.githubusercontent.com/5558824/33145749-60aeb370-cfa0-11e7-819a-c1f5245b1443.png)|![simulator screen shot - iphone 8 - 2017-11-22 at 16 14 41](https://user-images.githubusercontent.com/5558824/33145748-6086d47c-cfa0-11e7-8e96-8d126584803b.png)
